### ISSUE TITLE
Lesson 10-15, Gradient descent

### ISSUE THREAD

**Issue by kittle31** [udacity/dlnd-issue-reports#105]
The code in the solutions indicates the gradient depends on the error amount, when it really just a partial derivative.  This is a source of much confusion

**Comment by giladgressel** [comment]
it would be helpful if you provided a link for which solutions you were referring to

**Comment by kittle31** [comment]
See the solution for chapter 10 - line 24
https://classroom.udacity.com/nanodegrees/nd101/parts/2a9dba0b-28eb-4b0e-acfa-bdcf35680d90/modules/329a736b-1700-43d4-9bf0-753cc461bebc/lessons/dc37fa92-75fd-4d41-b23e-9659dde80866/concepts/7d480208-0453-4457-97c3-56c720c23a89

and then the solution for chapter 11 - line 36
https://classroom.udacity.com/nanodegrees/nd101/parts/2a9dba0b-28eb-4b0e-acfa-bdcf35680d90/modules/329a736b-1700-43d4-9bf0-753cc461bebc/lessons/dc37fa92-75fd-4d41-b23e-9659dde80866/concepts/4b167ce0-9d45-45e1-bfe6-891b2c68ac94

Each of the solutions express the "gradient" in terms of the errors.  it would be much clearer to make the gradient calculation and weight delta calculation into separate steps.

**Comment by clapollo** [comment]
I went through and tried to make some of this more consistent with what the new video says. (It uses the phrase 'error term' to refer to the same thing you're referring to in this issue.). However, people do usually refer to this as the error gradient, which is the output error multiplied by the derivative of the output activation function. I'm hoping it's clearer now with the new content that was added earlier to help explain back propagation and the few modifications I made.

If you'd like to check it out, I tweaked content in Gradient Descent, Gradient Descent: The Code, Implementing Gradient Descent, and Implementing Backpropagation. You'll need to reset your quizzes if you want to see the code changes that match the text changes.

I could also see separating the gradient calculation into it's own step, but I didn't want the code to stray too far from the text because the text matches up with the video.
